pvlib/pvsystem.py

```diff
@@ -1794,71 +1846,152 @@ def singlediode(photocurrent, saturation_current, resistance_series,
     [3] D. King et al, "Sandia Photovoltaic Array Performance Model",
     SAND2004-3535, Sandia National Laboratories, Albuquerque, NM
 
+    [4] "Computer simulation of the effects of electrical mismatches in
+    photovoltaic cell interconnection circuits" JW Bishop, Solar Cell (1988)
+    https://doi.org/10.1016/0379-6787(88)90059-2
+
     See also
     --------
     sapm
     calcparams_desoto
     '''
 
-    # Compute short circuit current
-    i_sc = i_from_v(resistance_shunt, resistance_series, nNsVth, 0.,
-                    saturation_current, photocurrent)
+    if method.lower() == 'lambertw':
+        # Compute short circuit current
+        i_sc = i_from_v(resistance_shunt, resistance_series, nNsVth, 0.,
+                        saturation_current, photocurrent, method)
+
+        # Compute open circuit voltage
+        v_oc = v_from_i(resistance_shunt, resistance_series, nNsVth, 0.,
+                        saturation_current, photocurrent, method)
+
+        params = {'r_sh': resistance_shunt,
+                  'r_s': resistance_series,
+                  'nNsVth': nNsVth,
+                  'i_0': saturation_current,
+                  'i_l': photocurrent}
+
+        p_mp, v_mp = _golden_sect_DataFrame(params, 0., v_oc * 1.14,
+                                            _pwr_optfcn)
+
+        # Invert the Power-Current curve. Find the current where the inverted
```

Comment:
Is this comment still relevant?

Reply:
This line is misplaced and in perhaps misworded. I think it belongs to the line above which finds the max power and voltage at max power. I've reworded and moved it.

Reply:
✔️